devices/Mcp23xxx/Mcp23xxx.cs

```diff
@@ -0,0 +1,572 @@
+        /// <returns>The byte address of the register for the given port bank and bank style.</returns>
+        private byte GetMappedAddress(Register register, Port port = Port.PortA,
+            BankStyle bankStyle = BankStyle.Sequential)
+        {
+            if (port != Port.PortA && port != Port.PortB)
+            {
+                throw new ArgumentOutOfRangeException(nameof(port));
+            }
+
+            if (bankStyle != BankStyle.Separated && bankStyle != BankStyle.Sequential)
+            {
+                throw new ArgumentOutOfRangeException(nameof(bankStyle));
+            }
+
+            byte address = (byte)register;
+
+            // There is no mapping for 8 bit expanders
+            if (PinCount == 8)
+            {
+                return address;
+            }
+
+            if (bankStyle == BankStyle.Sequential)
+            {
+                // Registers for each bank are sequential
+                // (IODIRA = 0x00, IODIRB = 0x01, IPOLA = 0x02, IPOLB = 0x03, ...)
+                address += address;
+                return port == Port.PortA ? address : ++address;
+            }
+
+            // Registers for each bank are separated
+            // (IODIRA = 0x00, ... OLATA = 0x0A, IODIRB = 0x10, ... OLATB = 0x1A)
+            return port == Port.PortA ? address : address += 0x10;
+        }
+
+        /// <inheritdoc/>
+        protected override void OpenPin(int pinNumber)
+        {
+            // No-op
+        }
+
+        /// <inheritdoc/>
+        protected override void ClosePin(int pinNumber)
+        {
+            // No-op
+        }
+
+        /// <inheritdoc/>
+        protected override PinMode GetPinMode(int pinNumber)
+        {
+            ValidatePin(pinNumber);
+
+            // IsBitSet returns true if bitNumber is flipped on in data.
+            bool IsBitSet(byte data, int bitNumber) => (data & (1 << bitNumber)) != 0;
+
+            if (pinNumber < 8)
+            {
+                return IsBitSet(InternalReadByte(Register.IODIR, Port.PortA), pinNumber)
+                    ? PinMode.Input
+                    : PinMode.Output;
+            }
+            else
+            {
+                return IsBitSet(InternalReadByte(Register.IODIR, Port.PortB), pinNumber - 8)
+                    ? PinMode.Input
+                    : PinMode.Output;
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventTypes,
+            PinChangeEventHandler callback) => throw new NotImplementedException();
+
+        /// <inheritdoc/>
+        protected override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback) =>
+            throw new NotImplementedException();
+
+        /// <inheritdoc/>
+        protected override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => pinNumber;
+
+        /*
```

Comment:
if you want to keep this one commented, please add a remark why

Reply:
removed